backend/satellite_tools/reposync.py

```diff
@@ -1130,13 +1130,20 @@ def import_packages(self, plug, source_id, url, is_non_local_repo):
             log(0, '')
             log(0, "  Linking packages to the channel.")
             # Packages to append to channel
-            import_batch = [self.associate_package(pack) for (pack, to_download, to_link) in to_process if to_link]
-            backend = SQLBackend()
-            caller = "server.app.yumreposync"
-            importer = ChannelPackageSubscription(import_batch,
-                                                  backend, caller=caller, repogen=False)
-            importer.run()
-            backend.commit()
+            import_batches = list(self.chunks(
+                [self.associate_package(pack) for (pack, to_download, to_link) in to_process if to_link],
+                1000))
+            count = 0
+            for import_batch in import_batches:
+                backend = SQLBackend()
```

Comment:
Do we need a new instance of backend every time?

Reply:
Well, I am unsure which data might be stay in the object any could destroy the run.
As this is tested I would keep it now and a later refactoring could try it.